hack/generator/pkg/astmodel/property_definition.go

```diff
@@ -22,7 +22,7 @@ type PropertyDefinition struct {
 	propertyName PropertyName
 	propertyType Type
 	description  string
-	validations  []Validation
+	isRequired   bool
```

Comment:
Isn't this about the required validation?

Why have this here then?

Reply:
Oh it's about the validation _and_ the `Optional`-ness of the type, ok.

Reply:
the requiredness is a property of the property, not of the type of the property (this is all a bit conflated by us because we tie this to optionalness via MakeRequired/MakeOptional)